crates/ruff_linter/src/linter.rs

```diff
@@ -113,6 +112,17 @@ pub fn check_path(
     parsed: &Parsed<ModModule>,
     target_version: TargetVersion,
 ) -> Vec<Message> {
+    let source_file = LazyCell::new(|| {
```

Comment:
I think the `LazyCell` here is mostly useless because you always dereference it when calling `check_tokens`, `check_file_path`. We should probably just build it eagerly